ignite/handlers/param_scheduler.py

```diff
@@ -852,6 +852,44 @@ def __init__(
                 f"torch.optim.lr_scheduler.{PyTorchLRScheduler.__name__}, "
                 f"but given {type(lr_scheduler)}"
             )
+        if isinstance(lr_scheduler, torch.optim.lr_scheduler.CosineAnnealingWarmRestarts):
+
+            class _CosineAnnealingWarmRestarts(CosineAnnealingWarmRestarts):
+                def get_lr(self, epoch: Union[int, None] = None) -> Union[List[float], float]:
+                    if epoch is None and self.last_epoch < 0:
+                        epoch = 0
+
+                    if epoch is None:
+                        epoch = self.last_epoch + 1
+                        self.T_cur = self.T_cur + 1
+                        if self.T_cur >= self.T_i:
+                            self.T_cur = self.T_cur - self.T_i
+                            self.T_i = self.T_i * self.T_mult
+                    else:
+                        if epoch < 0:
+                            raise ValueError("Expected non-negative epoch, but got {}".format(epoch))
+                        if epoch >= self.T_0:
+                            if self.T_mult == 1:
+                                self.T_cur = epoch % self.T_0
+                            else:
+                                n = int(math.log((epoch / self.T_0 * (self.T_mult - 1) + 1), self.T_mult))
+                                self.T_cur = epoch - self.T_0 * (self.T_mult**n - 1) / (self.T_mult - 1)
+                                self.T_i = self.T_0 * self.T_mult ** (n)
+                        else:
+                            self.T_i = self.T_0
+                            self.T_cur = epoch
+                    self.last_epoch = math.floor(epoch)
+
+                    return super(_CosineAnnealingWarmRestarts, self).get_lr()
```

Comment:
Let's define that outside the class, at the same level as other classes.
Let's make it as a wrapping class: `lr_scheduler = _CosineAnnealingWarmRestarts(lr_scheduler)`

Reply:
maybe add `from`  like this?
`lr_scheduler = _CosineAnnealingWarmRestarts.from_scheduler(lr_scheduler)`?

Reply:
why to not declare _CosineAnnealingWarmRestarts inside __init__? 
will we use these two classes (_CosineAnnealingWarmRestarts, LRScheduler) separately from each other?

Reply:
> why to not declare _CosineAnnealingWarmRestarts inside __init__?
> will we use these two classes (_CosineAnnealingWarmRestarts, LRScheduler) separately from each other?

No, but code becomes rather hard to read and bulky. Prepending `_` into the class means that it is private, so even defined in the global scope it does not mean that we publicly expose it.